modules/internal/DefaultRectangular.chpl

```diff
@@ -916,6 +946,7 @@ module DefaultRectangular {
                                            stridable=stridable);
     var off: rank*idxType;
     var blk: rank*chpl__idxTypeToIntIdxType(idxType);
+    var SizesPerDim: rank*chpl__idxTypeToIntIdxType(idxType);
```

Comment:
I'd expect it to start with a lower case letter, like `sizesPerDim`. I'd also like to know more about why adding this field is necessary. Is it mostly the same as blk? What if you just accessed dom.dsiDim(dim).length when you needed this information?

Reply:
From what I had discovered, polly mainly uses the row-major format for index computation(i.e for Arr[1..4][1..5][1..6], Arr[i][j][k] would correspond to the expression `6*(5*(i) + j) + k`. Since the blk values are precomputed coefficients of the indices in the index expression, it is difficult for Polly to allocate correct array sizes in the case of GPUCodegen, once the blk values are used instead of array dimension sizes.

Reply:
As far as the last question is concerned, I didn't want to generate extra GEP's and load instructions just for accessing directly from `dom`. Hence, I had precomputed these values in `postinit()` which can be used for further reference.